tests/v2/validation/rbac/globalrolesv2/globalroles_v2.go

```diff
@@ -21,44 +21,93 @@ import (
+	corev1 "k8s.io/api/core/v1"
 	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/selection"
 	kwait "k8s.io/apimachinery/pkg/util/wait"
+	"github.com/rancher/shepherd/extensions/kubeapi/namespaces"
+
 )
 
 const (
-	roleOwner          = "cluster-owner"
-	roleMember         = "cluster-member"
-	roleProjectOwner   = "project-owner"
-	roleCrtbView       = "clusterroletemplatebindings-view"
-	roleProjectsCreate = "projects-create"
-	roleProjectsView   = "projects-view"
-	standardUser       = "user"
-	localcluster       = "local"
-	crtbOwnerLabel     = "authz.management.cattle.io/grb-owner"
-	namespace          = "fleet-default"
-	localPrefix        = "local://"
-	clusterContext     = "cluster"
-	projectContext     = "project"
-	bindingLabel       = "membership-binding-owner"
+	roleOwner           = "cluster-owner"
+	roleMember          = "cluster-member"
+	roleProjectOwner    = "project-owner"
+	roleCrtbView        = "clusterroletemplatebindings-view"
+	roleProjectsCreate  = "projects-create"
+	roleProjectsView    = "projects-view"
+	standardUser        = "user"
+	localcluster        = "local"
+	ownerLabel      = "authz.management.cattle.io/grb-owner"
+	namespace           = "fleet-default"
+	localPrefix         = "local://"
+	clusterContext      = "cluster"
+	projectContext      = "project"
+	bindingLabel        = "membership-binding-owner"
+	globalDataNamespace = "cattle-global-data"
+	defaultNamespace    = "default"
 )
 
-var globalRole = v3.GlobalRole{
-	ObjectMeta: metav1.ObjectMeta{
-		Name: "",
-	},
-	InheritedClusterRoles: []string{},
-}
+var (
+	globalRole = v3.GlobalRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "",
+		},
+		InheritedClusterRoles: []string{},
+	}
 
-var globalRoleBinding = &v3.GlobalRoleBinding{
-	ObjectMeta: metav1.ObjectMeta{
-		Name: "",
-	},
-	GlobalRoleName: "",
-	UserName:       "",
-}
+	globalRoleBinding = &v3.GlobalRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "",
+		},
+		GlobalRoleName: "",
+		UserName:       "",
+	}
+
+	readSecretsPolicy = rbacv1.PolicyRule{
+		Verbs:     []string{"get", "list", "watch"},
+		APIGroups: []string{""},
+		Resources: []string{"secrets"},
+	}
+
+	readCRTBsPolicy = rbacv1.PolicyRule{
+		Verbs:     []string{"get", "list", "watch"},
+		APIGroups: []string{"management.cattle.io"},
+		Resources: []string{"clusterroletemplatebindings"},
+	}
+
+	readServiceAccountPolicy = rbacv1.PolicyRule{
+		Verbs:     []string{"get", "list", "watch"},
+		APIGroups: []string{""},
+		Resources: []string{"pods"},
+	}
+
+	readAllResourcesPolicy = rbacv1.PolicyRule{
+		Verbs:     []string{"get", "list", "watch"},
+		APIGroups: []string{"*"},
+		Resources: []string{"*"},
+	}
+
+	secret = corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namegen.AppendRandomString("secret-"),
+		},
+		Data: map[string][]byte{
+			"key": []byte(namegen.RandStringLower(5)),
+		},
+	}
+	project = &v3.Project{
```

Comment:
I do not have to use all the resources defined in the project template. are you suggesting to re use the project template that was defined in projects.go? Since thats a helper for projects test, I dont think its a good practice but yes if we can move it to actions or shepherd templates similar to https://github.com/rancher/shepherd/blob/main/extensions/projects/template.go.


I think we should update it to avoid repetions in every test that uses project template.

Reply:
You can use the project template that's available here: https://github.com/rancher/shepherd/blob/a83dc44ada4aebe1acaea2538b99bab314f5a5c8/extensions/kubeapi/projects/projects.go#L30